requirements/app/base.txt

```diff
@@ -13,6 +13,7 @@ inquirer >=2.10.0, <=3.1.3
 psutil <5.9.5
 click <=8.1.3
 python-multipart>=0.0.5, <=0.0.6
+backoff >=2.2.1, <2.3.0
```

Comment:
In your opinion, is the package sufficiently maintained? It seems it has no activity, which worries me. If it is a concern, one could reuse our logic in the LightningClient for exponential backoff too (https://github.com/Lightning-AI/lightning/blob/2d5964d7e122182de4521d571d0f43bb0a867731/src/lightning/app/utilities/network.py#L136)

Reply:
This is a good point! I totally forgot about the fact we had a decorator already. Let me test that instead of the dependency.

Reply:
Ok, our wrapper would need work as it is very specific to the LightningClient. I'll summarize why it's a good idea to adopt the library in the PR description.